### ISSUE TITLE
Connecting to Alexa from 3rd party Unity app

### ISSUE THREAD

**Issue by AbraTabia** [timheuer/alexa-skills-dotnet#256]
Hey guys, I have been setting up your library in to my Unity game, Unity is written in C# and its not exactly the same as most C# environments but I got it in to my project and my game builds on my firetv with no errors.

I think I understand a lot of concepts here, but one thing that is confusing me everywhere, is mixing up the difference between the lambda environment code and the code I am trying to add to my game. So can your library be used on the 3rd party app or is it just for the lambda server environment? I use my own lambda server inside the same game for my multiplayer, so am familiar with it but was letting Amazon host this first skill to save me the money as the alexa controller. I have the skill almost done in the console and most of the code done inside my game (intent handlers for all my game buttons etc) and I even have your library coded but I am not seeing anywhere to add my lambda url to, or alexa client id/secret to so I feel like I may have confused these 2 parts again.

I read about web games for alexa, my game is webgl but more for android/ios and PC, but if I could fetch the commands sent by alexa to my lambda function from my game, and send messages from my game to lambda (which would then send them to the alexa device) I could just call lambda as an api from my game.

So mainly I asking about using your library from Unity and if its not possible, any direction you can send me to the right place would be helpful.

Thanks,
Paul

**Comment by timheuer** [comment]
Hi @AbraTabia -- essentially this is a strongly-typed request/response library to make it easier to work with all the data that comes from skills and other subsequent actions, etc.  It really relies on someone providing the context of a skill request and also to be able to handle the response.  I'm not familiar with hosting your own lambda server, but you'll need that context.  This library is not intended to be a 'host' for the skills service though (hence, no place to put any alexa ID, etc.  It's expected to be hosted in some place that can communicate with Alexa communication.  Ideally you'd put your lambda URL in the Alexa Skills service endpoint information for your skill:
![image](https://user-images.githubusercontent.com/4821/202924130-a8c32967-8aa5-405a-9b73-2d14d890dc00.png)

**Comment by AbraTabia** [comment]
Hey Tim, thanks for the reply and ya that makes sense. Most solutions I found are the same, I liked yours better and hoped I could get it to work in Unity. And ya it does feel like I will have to use my own lambda server for this, I saw some options on linking the free account to mine so I will check out those options. I could write my own lambda, maybe even use this library since I am familiar with it now and I know I can get my game side to work the same way my multiplayer does, just have lambda store the intents in dynamoDB and the game poll lambda every few seconds looking for new information but that causes a slight delay and of course gets expensive.

Anyway I will figure that out, but I am glad I asked to know this won't work and I think nothing else prebuilt will either. I'll try to get the web games api which is in javascript to work in unity, it won't be pretty haha. I'll ask Amazon too.
